### PR TITLE
Follow up fixes with feedback from #74129

### PR DESCRIPTION
**Pull request by gottesmm**
Didn't need to modify any tests since this shouldn't have any functional change. Just a slight cleanup.

### CODE REVIEW

**Comment by gottesmm** [comment]
@xedin as per your request.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci test windows platform

File: lib/Sema/CSDiagnostics.cpp [hunk 1]
```diff
@@ -7918,21 +7918,25 @@ bool NonEphemeralConversionFailure::diagnoseAsError() {
   return true;
 }
 
-bool SendingOnFunctionParameterMismatchFail::diagnoseAsError() {
-  emitDiagnosticAt(getLoc(), diag::sending_function_wrong_sending,
-                   getFromType(), getToType())
+bool SendingMismatchFailure::diagnoseAsError() {
+  if (getLocator()->getLastElementAs<LocatorPathElt::FunctionArgument>())
```
Comment (xedin) [inline lib/Sema/CSDiagnostics.cpp:7922]: Nit: can use `isLastElement` instead of `getLastElementAs`.

File: include/swift/Sema/CSFix.h [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #include "swift/AST/Type.h"
 #include "swift/AST/Types.h"
 #include "swift/Basic/Debug.h"
+#include "swift/Sema/Constraint.h"
```
Comment (xedin) [inline include/swift/Sema/CSFix.h:27]: Do we need this include?

**Review by xedin — APPROVED** [review]

File: include/swift/Sema/CSFix.h (hunk 2 above)
Comment (gottesmm) [reply include/swift/Sema/CSFix.h:27]: Yes. Surprisingly.

**Pull request merged**